wisp/wisp-rate-limiting/src/main/kotlin/wisp/ratelimiting/RateLimiter.kt

```diff
@@ -27,13 +27,37 @@ interface RateLimiter {
     amount: Long = 1
   ): ConsumptionData
 
+  /**
+   * This tests whether [amount] tokens are available in the bucket associated with the given key.
+   * It is essentially a dry run of [consumeToken].
+   * Note that this data may be stale when it comes back, as time has elapsed and other pods
+   * could have taken tokens in the meantime.
+   */
+  fun testConsumptionAttempt(
+    key: String,
+    configuration: RateLimitConfiguration,
+    amount: Long = 1
+  ): TestConsumptionResult
+
   /**
    * Releases [amount] tokens back to the bucket associated with the given key
    * This will raise any exception thrown by the bucket4j proxy manager implementation, e.g.
    * subclasses of [JedisException] when using the Jedis implementation.
    */
   fun releaseToken(key: String, configuration: RateLimitConfiguration, amount: Long = 1)
 
+  /**
+   * Returns how many tokens remain in the bucket.
+   * Note that this data may be stale when it comes back, as time has elapsed and other pods
+   * could have taken tokens in the meantime.
+   */
+  fun availableTokens(key: String, configuration: RateLimitConfiguration): Long
+
+  /**
+   * Resets the bucket back to its maximum capacity
```

Comment:
Is this dangerous? What’s the use case for this?

Reply:
Some rate limits are used as mutexes, and we want to reset the bucket after they're used. There may also be other cases based on business logic (e.g. I can imagine a scenario where we want reset a customer's attempts for an action under some circumstance) but that's a hypothetical on my part